File: webapp/airTicket/src/app/core/service/air-port.service.ts
import {Injectable} from '@angular/core';
import {CoreModule} from '../core.module';
import {AirPort} from '../../entity/AirPort';
import {Page} from '../../entity/norm/page';
import {Pageable} from '../../entity/norm/pageable';
import {Observable, of} from 'rxjs';
import {City} from '../../entity/City';
import {HttpClient} from '@angular/common/http';

@Injectable({
  providedIn: CoreModule
})
export class AirPortService {

  baseUrl = 'api/airPort';

  constructor(private httpClient: HttpClient) {
  }

  // 获取所有的分页
  pageByName(name: string, pageable: Pageable): Observable<Page<AirPort>> {
    const params = {
      page: pageable.page.toString(),
      size: pageable.size.toString(),
      name
    };
    return this.httpClient.get<Page<AirPort>>(this.baseUrl + '/pageByName', {params});
  }

  // 通过id获取
  getById(id: number): Observable<AirPort> {
    return this.httpClient.get<AirPort>(this.baseUrl + '/' + id);
  }

  // 保存
  save(airPort: AirPort): Observable<AirPort> {
    return this.httpClient.post<AirPort>(this.baseUrl, airPort);
  }

  // 更新
  update(id: number, airPort: AirPort): Observable<AirPort> {
    return this.httpClient.put<AirPort>(this.baseUrl + '/' + id, airPort);
  }

  // 删除
  delete(id: number): Observable<any> {
    return this.httpClient.delete(this.baseUrl + '/' + id);
  }
}

File: api/airTicket/src/main/java/com/xiang/airTicket/service/AirPortService.java
package com.xiang.airTicket.service;

import com.xiang.airTicket.entity.AirPort;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface AirPortService {

    // 保存
    AirPort save(AirPort airPort);

    // 根据名称获取分页信息
    Page<AirPort> pageByName(String name, Pageable pageable);

    // 编辑
    AirPort update(Long id, AirPort airPort);

    // 通过id获取
    AirPort getById(Long id);

    // 删除
    void delete(Long id);
}
